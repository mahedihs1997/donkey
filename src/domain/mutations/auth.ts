import gql from 'graphql-tag';

export const LOGIN_MUTATION = gql`
  mutation Login($username: String!, $password: String!) {
    signIn(login: $username, password: $password) {
      token
    }
  }
`;
