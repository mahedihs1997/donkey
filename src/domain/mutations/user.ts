import gql from 'graphql-tag';

export const CREATE_USER_QUERY = gql`
  mutation CreateUser($email: String!, $password: String!) {
    loginUser(data: { email: $email, password: $password }) {
      token
    }
  }
`;
