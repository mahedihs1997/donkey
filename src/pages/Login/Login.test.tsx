import { MockedProvider } from '@apollo/react-testing';
import { LOGIN_MUTATION } from '@domain/mutations/auth';
import React from 'react';
import TestRenderer from 'react-test-renderer';

import LoginPage from '.';

const mocks = [
  {
    request: {
      query: LOGIN_MUTATION,
      variables: {
        email: 'Ugendo@buugendo.com',
        password: '123',
      },
    },
    result: {
      data: {
        token: 'dont-look-bro',
      },
    },
  },
];

describe('<LoginPage />', () => {
  it('should render without throwing an error and match snapshot', () => {
    const testRenderer = TestRenderer.create(
      <MockedProvider addTypename={false} mocks={mocks}>
        <LoginPage />
      </MockedProvider>
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
