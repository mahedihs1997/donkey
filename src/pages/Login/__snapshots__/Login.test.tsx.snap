// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> should render without throwing an error and match snapshot 1`] = `
<div
  className="sc-AxjAm SaElU"
>
  <h3
    className="MuiTypography-root MuiTypography-h3"
  >
    Welcome back!
  </h3>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-input-1 MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        Email
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="username"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </p>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-input-1 MuiFormControl-fullWidth"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
        data-shrink={false}
      >
        Password
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="password"
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          name="password"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="password"
          value=""
        />
      </div>
      <p
        className="MuiFormHelperText-root"
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </p>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Login
      </span>
    </button>
  </form>
</div>
`;
